File: src/main/java/gruvexp/bbminigames/listeners/ShiftListener.java
package gruvexp.bbminigames.listeners;

import gruvexp.bbminigames.twtClassic.BotBows;
import gruvexp.bbminigames.twtClassic.BotBowsPlayer;
import gruvexp.bbminigames.twtClassic.Lobby;
import org.bukkit.GameMode;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerToggleSneakEvent;

public class ShiftListener implements Listener {

    @EventHandler
    public void onShiftToggle(PlayerToggleSneakEvent e) {
        Player p = e.getPlayer();
        Lobby lobby = BotBows.getLobby(p);
        if (lobby == null) return;
        if (!lobby.isGameActive()) {return;}
        if (p.getGameMode() != GameMode.ADVENTURE) {return;}
        BotBowsPlayer bp = lobby.getBotBowsPlayer(p);

        if (bp.isSneakingExhausted() && e.isSneaking()) {
            BotBows.debugMessage("its not allowed to sneak");
            e.setCancelled(true);
        }
    }
}

File: src/main/java/gruvexp/bbminigames/listeners/ShiftListener.kt
package gruvexp.bbminigames.listeners

import gruvexp.bbminigames.twtClassic.BotBows
import net.kyori.adventure.text.Component
import net.kyori.adventure.text.format.NamedTextColor
import org.bukkit.GameMode
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerToggleSneakEvent

class ShiftListener : Listener {
    @EventHandler
    fun onShiftToggle(e: PlayerToggleSneakEvent) {
        val p = e.getPlayer()
        val bp = BotBows.getBotBowsPlayer(p) ?: return
        if (!bp.lobby.isGameActive) return
        if (p.gameMode != GameMode.ADVENTURE) return


        if (bp.isSneakingExhausted && e.isSneaking) {
            p.sendActionBar(Component.text("You are exhausted and cant sneak", NamedTextColor.RED))
            e.isCancelled = true
        }
    }
}
